src/account/account.js

```diff
@@ -0,0 +1,119 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+// @flow
+
+import request from "request-promise-native"
+
+import { MODERN_CIPHER_SUITES } from "../utils"
+import isToken from "../utils/is-token"
+
+// Necessary to make gRPC work
+process.env.GRPC_SSL_CIPHER_SUITES = MODERN_CIPHER_SUITES
+
+/**
+ * `Account` is a client for The Things Network account server API.
+ * It can be used to manage applications and their EUIs, as well as gateways.
+ * Either a Bearer Token or an Application Access Key can be used for
+ * authentication. The latter method allows to use the `getApplication()`
+ * function only.
+ *
+ * Example:
+ * ```
+ * const account = new Account("accesKeyOrToken", "https://customserveradress.org")
+ * ```
+ */
+export class AccountClient {
+  /** @private */
+  authHeader : string
+
+  /** @private */
+  serverAddress: string
+
+  constructor (accessKeyOrToken : string, serverAddress : string = "https://account.thethingsnetwork.org") : void {
+    this.authHeader = isToken(accessKeyOrToken)
+      ? `Bearer ${accessKeyOrToken}`
+      : `Key ${accessKeyOrToken}`
+
+    this.serverAddress = serverAddress
+  }
+
+  /** @private */
+  async makeRequest (url : string, method : string = "GET", body : ?any) : Promise<any> {
+    return await request({
+      method,
+      url: `${this.serverAddress}/${url}`,
+      headers: { Authorization: this.authHeader },
+      json: true,
+      body,
+    })
+  }
+
+  async getAllApplications () : Promise<Array<Application>> {
```

Comment:
There's no need to  wrap this in an `async/await` since we get a promise already from `makeRequest`.